server/app/controllers/task-controller.js

```diff
@@ -0,0 +1,102 @@
+const taskService = require('../services/task-service');
+const kommissionService = require('../services/kommission-service');
+const { Task } = require('../models');
+const { TaskSchema } = require('../models/schemas');
+const { createUserError } = require('../../utils');
+
+/**
+ * Create a task.
+ *
+ * @param req Request
+ * @param res Response
+ * @return {Promise.<void>} Nothing
+ */
+async function createTask(req, res) {
+    const schema = TaskSchema.requiredKeys(
+        'name',
+        'deadline',
+        'state',
+        '_embedded',
+        '_embedded.kommissionId',
+    );
+
+    const { error } = schema.validate(req.body);
+    if (error) throw createUserError('BadRequest', error.message);
+
+    await kommissionService.getKommissionById(req.body._embedded.kommissionId);
+
+    let newTask = new Task({
+        ...req.body,
+        _embedded: undefined // Remove the only external object
+    });
+
+    newTask = await taskService.createTask(newTask, req.body._embedded);
+
+    res.json(newTask);
+}
+
+
+/**
+ * Get a task by its id.
+ *
+ * @param req Request
+ * @param res Response
+ * @return {Promise.<void>} Nothing
+ */
+async function getTaskById(req, res) {
+    const taskId = req.params.id;
+
+    const task = await taskService.getTaskById(taskId);
+
+    res.json(task);
+}
+
+
+/**
+ * Update a task.
+ *
+ * @param req Request
+ * @param res Response
+ * @return {Promise.<void>} Nothing
+ */
+async function updateTask(req, res) {
+    const schema = TaskSchema.min(1);
+
+    const { error } = schema.validate(req.body);
+    if (error) throw createUserError('BadRequest', error.message);
+
+    if (req.body._embedded && req.body._embedded.kommissionId) await kommissionService.getKommissionById(req.body._embedded.kommissionId);
```

Comment:
What is this for??

If it's just ot check if the kommission exists, it should be done in the service not here

Reply:
Yes it is, so i'm going to move it into service